// @awesome-cordova-plugins/background-fetch/__ivy_ngcc__/ngx/index.js
import { __extends } from "tslib";
import { AwesomeCordovaNativePlugin, cordova } from '@awesome-cordova-plugins/core';
import { Injectable } from '@angular/core';
import * as ɵngcc0 from '@angular/core';
var BackgroundFetch = /** @class */ (function (_super) {
    __extends(BackgroundFetch, _super);
    function BackgroundFetch() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    BackgroundFetch.prototype.configure = function (config) { return cordova(this, "configure", { "callbackOrder": "reverse" }, arguments); };
    BackgroundFetch.prototype.start = function () { return cordova(this, "start", {}, arguments); };
    BackgroundFetch.prototype.stop = function () { return cordova(this, "stop", {}, arguments); };
    BackgroundFetch.prototype.finish = function (taskId) { return cordova(this, "finish", { "sync": true }, arguments); };
    BackgroundFetch.prototype.status = function () { return cordova(this, "status", {}, arguments); };
    BackgroundFetch.pluginName = "BackgroundFetch";
    BackgroundFetch.plugin = "cordova-plugin-background-fetch";
    BackgroundFetch.pluginRef = "BackgroundFetch";
    BackgroundFetch.repo = "https://github.com/transistorsoft/cordova-plugin-background-fetch";
    BackgroundFetch.platforms = ["iOS"];
BackgroundFetch.ɵfac = /*@__PURE__*/ function () { var ɵBackgroundFetch_BaseFactory; return function BackgroundFetch_Factory(t) { return (ɵBackgroundFetch_BaseFactory || (ɵBackgroundFetch_BaseFactory = ɵngcc0.ɵɵgetInheritedFactory(BackgroundFetch)))(t || BackgroundFetch); }; }();
BackgroundFetch.ɵprov = /*@__PURE__*/ ɵngcc0.ɵɵdefineInjectable({ token: BackgroundFetch, factory: function (t) { return BackgroundFetch.ɵfac(t); } });
(function () { (typeof ngDevMode === "undefined" || ngDevMode) && ɵngcc0.ɵsetClassMetadata(BackgroundFetch, [{
        type: Injectable
    }], null, null); })();
    return BackgroundFetch;
}(AwesomeCordovaNativePlugin));
export { BackgroundFetch };
